source/features/repo-header-info.tsx

```diff
@@ -0,0 +1,51 @@
+import * as pageDetect from 'github-url-detection';
+import {StarFillIcon} from '@primer/octicons-react';
+import React from 'dom-chef';
+import {CachedFunction} from 'webext-storage-cache';
+
+import observe from '../helpers/selector-observer.js';
+import features from '../feature-manager.js';
+import api from '../github-helpers/api.js';
+import GetRepositoryStargazerCount from './repo-header-info.gql';
+import abbreviateNumber from '../helpers/abbreviate-number.js';
+import {cacheByRepo} from '../github-helpers/index.js';
+
+const stargazerCount = new CachedFunction('stargazer-count', {
+	async updater(): Promise<number> {
+		const {repository: {stargazerCount}} = await api.v4(GetRepositoryStargazerCount);
+		return stargazerCount;
+	},
+	maxAge: {days: 1},
+	staleWhileRevalidate: {days: 3},
+	cacheKey: cacheByRepo,
+});
+
+async function add(navigationList: HTMLUListElement): Promise<void> {
+	navigationList.append(
+		<li className="color-fg-muted pl-2">
+			<div className="d-flex flex-items-center flex-justify-center ml-2 gap-1">
+				<StarFillIcon className="v-align-text-bottom"/>
+				<span className="Counter v-align-bottom">{abbreviateNumber(await stargazerCount.get())}</span>
+			</div>
+		</li>,
+	);
+}
+
+function init(signal: AbortSignal): void {
+	observe('header .AppHeader-context-full > nav ul', add, {signal});
```

Comment:
I'd make sure the icon appears before `ci-link` because `ci-link` is always delayed while this feature might be cached and instant.